### ISSUE TITLE
Icons are not been generated in Angular 8

### ISSUE THREAD

**Issue by dvelazdiaz** [jantimon/favicons-webpack-plugin#167]
I would like to use this plugin in Angular 8. I'm using the `@angular-builders/custom-webpack` package to create a partial webpack configuration. 

```
const FaviconsWebpackPlugin = require('favicons-webpack-plugin');

module.exports = {
  plugins: [
    new FaviconsWebpackPlugin()
  ]
};
```

When I try to generate a build the plugin is not creating the favicons. Do you have any idea of how I can solve this issue?

Thanks in advance

**Comment by jantimon** [comment]
Hey @dvelazdiaz - do you mean webpack-dev-server or production mode?

Where did you specify which png/svg to use?

**Comment by dvelazdiaz** [comment]
Both (webpack-dev-server and production). My project was bootstrapped with angular CLI, so I have the .png inside `src/assets/images`

**Comment by spencerspear** [comment]
I'm experiencing a similar issue. In development (using webpack dev server), the plugin only generates a single link tag.
```
<link rel="icon" href="/assets/favicon.png">
```

In a production build, I get the expected output added to the index.html file.
```
<link rel="shortcut icon" href="/assets/favicon.ico">
<link rel="icon" type="image/png" sizes="16x16" href="/assets/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="/assets/favicon-32x32.png">
<link rel="manifest" href="/assets/manifest.json">
<meta name="mobile-web-app-capable" content="yes">
<meta name="theme-color" content="#fff">
<meta name="application-name" content="gogodispatch">
<link rel="apple-touch-icon" sizes="57x57" href="/assets/apple-touch-icon-57x57.png">
<link rel="apple-touch-icon" sizes="60x60" href="/assets/apple-touch-icon-60x60.png">
<link rel="apple-touch-icon" sizes="72x72" href="/assets/apple-touch-icon-72x72.png">
<link rel="apple-touch-icon" sizes="76x76" href="/assets/apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" sizes="114x114" href="/assets/apple-touch-icon-114x114.png">
<link rel="apple-touch-icon" sizes="120x120" href="/assets/apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" sizes="144x144" href="/assets/apple-touch-icon-144x144.png">
<link rel="apple-touch-icon" sizes="152x152" href="/assets/apple-touch-icon-152x152.png">
<link rel="apple-touch-icon" sizes="167x167" href="/assets/apple-touch-icon-167x167.png">
<link rel="apple-touch-icon" sizes="180x180" href="/assets/apple-touch-icon-180x180.png">
<link rel="apple-touch-icon" sizes="1024x1024" href="/assets/apple-touch-icon-1024x1024.png">
<meta name="apple-mobile-web-app-capable" content="yes">
<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
<meta name="apple-mobile-web-app-title" content="gogodispatch">
<link rel="apple-touch-startup-image" media="(device-width: 320px) and (device-height: 480px) and (-webkit-device-pixel-ratio: 1)" href="/assets/apple-touch-startup-image-320x460.png">
<link rel="apple-touch-startup-image" media="(device-width: 320px) and (device-height: 480px) and (-webkit-device-pixel-ratio: 2)" href="/assets/apple-touch-startup-image-640x920.png">
<link rel="apple-touch-startup-image" media="(device-width: 320px) and (device-height: 568px) and (-webkit-device-pixel-ratio: 2)" href="/assets/apple-touch-startup-image-640x1096.png">
<link rel="apple-touch-startup-image" media="(device-width: 375px) and (device-height: 667px) and (-webkit-device-pixel-ratio: 2)" href="/assets/apple-touch-startup-image-750x1294.png">
<link rel="apple-touch-startup-image" media="(device-width: 414px) and (device-height: 736px) and (orientation: landscape) and (-webkit-device-pixel-ratio: 3)" href="/assets/apple-touch-startup-image-1182x2208.png">
<link rel="apple-touch-startup-image" media="(device-width: 414px) and (device-height: 736px) and (orientation: portrait) and (-webkit-device-pixel-ratio: 3)" href="/assets/apple-touch-startup-image-1242x2148.png">
<link rel="apple-touch-startup-image" media="(device-width: 768px) and (device-height: 1024px) and (orientation: landscape) and (-webkit-device-pixel-ratio: 1)" href="/assets/apple-touch-startup-image-748x1024.png">
<link rel="apple-touch-startup-image" media="(device-width: 768px) and (device-height: 1024px) and (orientation: portrait) and (-webkit-device-pixel-ratio: 1)" href="/assets/apple-touch-startup-image-768x1004.png">
<link rel="apple-touch-startup-image" media="(device-width: 768px) and (device-height: 1024px) and (orientation: landscape) and (-webkit-device-pixel-ratio: 2)" href="/assets/apple-touch-startup-image-1496x2048.png">
<link rel="apple-touch-startup-image" media="(device-width: 768px) and (device-height: 1024px) and (orientation: portrait) and (-webkit-device-pixel-ratio: 2)" href="/assets/apple-touch-startup-image-1536x2008.png">
<link rel="icon" type="image/png" sizes="228x228" href="/assets/coast-228x228.png">
<meta name="msapplication-TileColor" content="#fff">
<meta name="msapplication-TileImage" content="/assets/mstile-144x144.png">
<meta name="msapplication-config" content="/assets/browserconfig.xml">
<link rel="yandex-tableau-widget" href="/assets/yandex-browser-manifest.json">
```

## Configuration:
```
new FaviconsWebpackPlugin({
    logo: '/absolute/path/to/favicon.png'
}),
```

**Comment by jantimon** [comment]
@spencerspear this can be configured using the `mode` property

**Comment by spencerspear** [comment]
Thanks @jantimon! Setting `devMode: 'webapp'` sounds like it will do the trick.

**Comment by bzabor** [comment]
Yes thanks @jantimon! So helpful!

**Comment by HansCronau** [comment]
I believe this isssue by @dvelazdiaz got unintentionally hijacked by @spencerspear 's unrelated issue.

@dvelazdiaz posted his question to [StackOverflow](https://stackoverflow.com/questions/58018737/how-to-integrate-favicons-webpack-plugin-in-angular-8-projects) as well, where it got a reply a month after this issue was closed. User Volodya Senchak there noted that Angular 8 had stopped using `HtmlWebpackPlugin` for generating the `index.html` file. He mentioned that this could be circumvented by setting `"index": ""` in `angular.json`, but that this was an imperfect solution.

I have now run into the same issue, which seems to affect more than just `FaviconsWebpackPlugin`. Without the aforementioned fix, I cannot inject anything in the `index.html` from our `custom-webpack.config.js`, including the `HtmlWebpackPlugin` [options](https://github.com/jantimon/html-webpack-plugin#options) `title` and `base`.

There must be more Angular 8+ developers using `FaviconsWebpackPlugin`. I would be very interested to their thoughts on / solutions for the issue, as well as those of @jantimon and other maintainers.

**Comment by kontrollanten** [comment]
@HansCronau Did you find any solution?

**Comment by HansCronau** [comment]
@kontrollanten No. As far as I know this issue has not been resolved and was closed by mistake.  
We have worked around it by letting FaviconsWebpackPlugin write its output to a secondary dummy `index_dummy.html` and manually copying the relevant lines to our primary `index.html`. Downside: this takes the html injection out of the build process. This process thus needs to be repeated manually every time the icons change. Luckily for us this is rare.
